utils/build-script-impl

```diff
@@ -1001,138 +997,216 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
+            
+            SWIFT_SDKS+=("${swift_sdk}")
+
+            if [[ "${BUILD_SWIFT_STDLIB_UNITTEST_EXTRA}" == "1" ]] ; then
+                SWIFT_STDLIB_TARGETS+=("swift-stdlib-${stdlib_deployment_target}")
+            else
+                if [[ "${VALIDATION_TEST}" == "1" || "${LONG_TEST}" == "1" ]] ; then
+                    SWIFT_STDLIB_TARGETS+=("swift-stdlib-${stdlib_deployment_target}")
+                else
+                    SWIFT_STDLIB_TARGETS+=("swift-test-stdlib-${stdlib_deployment_target}")
+                fi
+            fi
         fi
-    fi
-    if [[ "${test_this_target}" ]] ; then
-        test_target_suffix=""
-        if [[ -n "${test_host_only}" ]] ; then
-            test_target_suffix="-non-executable"
+        if [[ "${build_benchmark_this_target}" ]] ; then
+            SWIFT_BENCHMARK_TARGETS+=("swift-benchmark-${stdlib_deployment_target}")
+            if [[ $(not ${SKIP_TEST_BENCHMARK}) ]] ; then
+              SWIFT_RUN_BENCHMARK_TARGETS+=("check-swift-benchmark-${stdlib_deployment_target}")
+            fi
         fi
+        if [[ "${test_this_target}" ]] ; then
+            test_target_suffix=""
+            if [[ -n "${test_host_only}" ]] ; then
+                test_target_suffix="-non-executable"
+            fi
 
-        test_subset_target_suffix=""
-        if [[ "${VALIDATION_TEST}" == "1" ]] ; then
-            if [[ "${LONG_TEST}" == "1" ]] ; then
-                test_subset_target_suffix="-all"
+            test_subset_target_suffix=""
+            if [[ "${VALIDATION_TEST}" == "1" ]] ; then
+                if [[ "${LONG_TEST}" == "1" ]] ; then
+                    test_subset_target_suffix="-all"
+                else
+                    test_subset_target_suffix="-validation"
+                fi
             else
-                test_subset_target_suffix="-validation"
+                if [[ "${LONG_TEST}" == "1" ]] ; then
+                    test_subset_target_suffix="-only_long"
+                fi
             fi
-        else
-            if [[ "${LONG_TEST}" == "1" ]] ; then
-                test_subset_target_suffix="-only_long"
+            SWIFT_TEST_TARGETS+=("check-swift${test_subset_target_suffix}${test_target_suffix}-${stdlib_deployment_target}")
+            if [[ $(not ${SKIP_TEST_OPTIMIZED}) && ! -n "${test_host_only}" ]] ; then
+                SWIFT_TEST_TARGETS+=("check-swift${test_subset_target_suffix}-optimize-${stdlib_deployment_target}")
             fi
         fi
-        SWIFT_TEST_TARGETS+=("check-swift${test_subset_target_suffix}${test_target_suffix}-${deployment_target}")
-        if [[ $(not ${SKIP_TEST_OPTIMIZED}) && ! -n "${test_host_only}" ]] ; then
-            SWIFT_TEST_TARGETS+=("check-swift${test_subset_target_suffix}-optimize-${deployment_target}")
-        fi
+    done
+
+    # Filter duplicate SWIFT_SDKs (which will exist if building for multiple architectures)
+    SWIFT_SDKS=(`echo "${SWIFT_SDKS[@]}" | tr " " "\n" | sort -u | tr "\n" " "`)
```

Comment:
@rintaro Exactly, `--swift-sdks` is for configuration.

As for why,

> The reason why this option is needed is that we build Swift in environments where a certain SDK (that we don't need to build for) appears to be available and would be detected and configured, but is actually unusable and would fail the build.